Bottom: f90154a6653c6669ada47d8dd8193f7e9ef97ac9
Top:    4073f245f45654701e392659ddd9170ea7d1fc8a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-19 00:59:32 +0000

action.c: Fixup some syntax so it appears as legal C before macro expansion

"if TEST_FLAG (...) {" does not look like legal C until the TEST_FLAG macro
is expanded. This works because the macro is wrapped with a () like all
good function-like pre-processor macros.

This bad style was upsetting cppcheck, which flagged up a syntax error for
this line. (Obviously it was not expanding macros before checking).


---

diff --git a/src/action.c b/src/action.c
index 2a0be9e..c1325ba 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2679,8 +2679,7 @@ ActionDisplay (int argc, char **argv, Coord childX, Coord childY)
 
 	case F_CycleClip:
 	  notify_crosshair_change (false);
-	  if TEST_FLAG
-	    (ALLDIRECTIONFLAG, PCB)
+	  if (TEST_FLAG (ALLDIRECTIONFLAG, PCB))
 	    {
 	      TOGGLE_FLAG (ALLDIRECTIONFLAG, PCB);
 	      PCB->Clipping = 0;
